concrete/controllers/single_page/dashboard/system/registration/automated_logout.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Concrete\Controller\SinglePage\Dashboard\System\Registration;
+
+use Carbon\Carbon;
+use Concrete\Core\Config\Repository\Repository;
+use Concrete\Core\Http\ResponseFactory;
+use Concrete\Core\Page\Controller\DashboardPageController;
+use Concrete\Core\Page\Page;
+use Concrete\Core\Url\Resolver\Manager\ResolverManager;
+
+class AutomatedLogout extends DashboardPageController
+{
+
+    const ITEM_IP = 'concrete.security.session.invalidate_on_ip_mismatch';
+    const ITEM_USER_AGENT = 'concrete.security.session.invalidate_on_user_agent_mismatch';
+    const ITEM_SESSION_INVALIDATE = 'concrete.session.valid_since';
+
+    /**
+     * The response factory we use to generate responses
+     *
+     * @var ResponseFactory
+     */
+    protected $factory;
+
+    /**
+     * The config repsository we save to
+     *
+     * @var \Concrete\Core\Config\Repository\Repository
+     */
+    protected $config;
+
+    /**
+     * The URL manager that tells us how to get to the page we're looking for
+     *
+     * @var \Concrete\Core\Url\Resolver\Manager\ResolverManager
+     */
+    protected $urls;
+
+    public function __construct(Page $c, ResponseFactory $factory, Repository $config, ResolverManager $urls)
+    {
+        parent::__construct($c);
+
+        $this->factory = $factory;
+        $this->config = $config;
+        $this->urls = $urls;
+    }
+
+    /**
+     * Main view function for the controller. This method is called when no other action matches the request
+     *
+     * @return \Concrete\Core\Http\Response|void
+     */
+    public function view()
+    {
+        $this->set('trustedProxyUrl', $this->urls->resolve(['/dashboard/system/permissions/trusted_proxies']));
+        $this->set('invalidateOnIPMismatch', $this->config->get(self::ITEM_IP));
+        $this->set('invalidateOnUserAgentMismatch', $this->config->get(self::ITEM_USER_AGENT));
+
+        $this->set('saveAction', $this->action('save'));
+        $this->set('invalidateAction', $this->action('invalidate_sessions', $this->token->generate('invalidate_sessions')));
+    }
+
+    /**
+     * An action for saving the Session Security form
+     * This method will manage saving settings and redirecting to the appropriate results page
+     *
+     * @return \Concrete\Core\Http\ResponseFactory|\Symfony\Component\HttpFoundation\Response
+     */
+    public function save()
+    {
+        if (!$this->token->validate('save_automated_logout')) {
+            $this->error->add($this->token->getErrorMessage());
+
+            $this->flash('error', $this->error);
```

Comment:
What's the benefit of using this, instead of `$this->flash($this->token->getErrorMessage())`?

Reply:
The thinking is that if I avoid using the object that is for tracking errors, I  risk making it more challenging for users to override / implement their own errors. Not high value here but I think it's important to use the errorlist object where possible